module/applications/actor/character-sheet.mjs

```diff
@@ -227,8 +227,14 @@ export default class ActorSheet5eCharacter extends ActorSheet5e {
       case "rollDeathSave":
         return this.actor.rollDeathSave({event: event});
       case "rollInitiative":
-        return this.actor.rollInitiative({createCombatants: true});
+        const initiative = new Promise((resolve) => {
+          resolve(this.actor.rollSkill("ini"));
+        });
+        let z = (await initiative);
+        let x = this.actor.rollInitiative({ createCombatants: true, initiativeOptions: { formula: (""+ z +"+"+ this.actor.system.attributes.init.bonus) } });
+        return (x);
```

Comment:
This section first calls a hidden initiative skill roll that builds the formula. Then passes that as a formula to the rollInitiative function to handle the rest.

Reply:
Hm, I'm not sure this is the correct approach to reusing the skill-rolling logic for initiative calculation. There are effects (like Bard's Jack of All Trades) that affect all skills, but don't affect initiative.

Reply:
If you don't mind me butting in, it would be more correct to be simply a Dexterity check, since that is how it is determined as per the Basic Rules.

> At the beginning of every combat, you roll initiative by making a Dexterity check. Initiative determines the order of creatures' turns in combat.

However I think it deserves to rather get the Death Saving Throw treatment. Like a Death Saving Throw is a special kind of Saving Throw, Initiative is a special kind of Ability Check, and constructing it fully without use of helpers from skills would likely end up being a lot cleaner in the end.

It would allow for roll hooks as well, e.g., `dnd5e.preRollInitiative` and `dnd5e.rollInitiative` just like there's `dnd5e.preRollDeathSave`.

Reply:
> If you don't mind me butting in, it would be more correct to be simply a Dexterity check, since that is how it is determined as per the Basic Rules.
> 
> > At the beginning of every combat, you roll initiative by making a Dexterity check. Initiative determines the order of creatures' turns in combat.
> 
> However I think it deserves to rather get the Death Saving Throw treatment. Like a Death Saving Throw is a special kind of Saving Throw, Initiative is a special kind of Ability Check, and constructing it fully without use of helpers from skills would likely end up being a lot cleaner in the end.
> 
> It would allow for roll hooks as well, e.g., `dnd5e.preRollInitiative` and `dnd5e.rollInitiative` just like there's `dnd5e.preRollDeathSave`.

The reason this is like this is that WOTC has indicated things like proficiency in Initiative are going to be a thing in later books. This allows for that.

Reply:
> If you don't mind me butting in, it would be more correct to be simply a Dexterity check, since that is how it is determined as per the Basic Rules.
> 
> > At the beginning of every combat, you roll initiative by making a Dexterity check. Initiative determines the order of creatures' turns in combat.
> 
> However I think it deserves to rather get the Death Saving Throw treatment. Like a Death Saving Throw is a special kind of Saving Throw, Initiative is a special kind of Ability Check, and constructing it fully without use of helpers from skills would likely end up being a lot cleaner in the end.
> 
> It would allow for roll hooks as well, e.g., `dnd5e.preRollInitiative` and `dnd5e.rollInitiative` just like there's `dnd5e.preRollDeathSave`.

Furthermore, this allows a few things. Example: this allows any skill to be used when rolling initiative which is a large upgrade, it also allows advantage and disadvantage rolls and roll bonuses, which is important for things like gift of alacrity.

Reply:
(Assuming you meant ability, not skill) Isn't that going to be set on the sheet, not in the roll dialog? Plus death saves allow for proficiency, bonuses, and advantage/disadvantage in their own method, I think that should be doable for Initiative as well in the same way.

Reply:
> (Assuming you meant ability, not skill) Isn't that going to be set on the sheet, not in the roll dialog? Plus death saves allow for proficiency, bonuses, and advantage/disadvantage in their own method, I think that should be doable for Initiative as well in the same way.

This also makes it very easy to interact with it for future changes.